### PR TITLE
Unit tests for tools.py

### PR DESCRIPTION
**Pull request by ayushrawat6367**
Updated tests for Ragpipeline and Quizbuilder classes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,13 +6,19 @@ COPY requirements.txt /code/requirements.txt
...
 # Local development key set
 # ENV TYPES: dev, production
 # When set to dev, API Key on endpoint requests are just 'dev'
 # When set to production, API Key on endpoint requests are the actual API Key
 
-ENV PYTHONPATH=/code/app
+
+ENV GOOGLE_APPLICATION_CREDENTIALS=/app/local-auth.json 
+ENV ENV_TYPE="dev"
+ENV PROJECT_ID="heroic-climber-423918-r1"
```
Comment (hash2004) [inline Dockerfile:19]: Hard coding ENV variables is not best practice.

File: app/api/router.py [hunk 2]
```diff
@@ -19,13 +19,17 @@ def read_root():
 async def submit_tool( data: ToolRequest, _ = Depends(key_check)):     
     try: 
         # Unpack GenericRequest for tool data
+        print("Before request_data")
         request_data = data.tool_data
         
+        print("Before load_tool_metadata")
         requested_tool = load_tool_metadata(request_data.tool_id)
-        
+        print("Before finalize_inputs")
```
Comment (hash2004) [inline app/api/router.py:27]: Use logger for these types of print statements.

File: app/api/tests/test_tools.py [hunk 3]
```diff
@@ -10,22 +10,25 @@
...
-    if old_env_type:
-        os.environ['ENV_TYPE'] = old_env_type
-    if old_project_id:
-        os.environ['PROJECT_ID'] = old_project_id
+    os.environ['GOOGLE_API_KEY'] = os.getenv('GOOGLE_API_KEY')
+
+    print('@@@@@ ENV')
+    print(os.environ['GOOGLE_API_KEY'])
+    # os.environ['ENV_TYPE'] = 'dev'
+    # os.environ['PROJECT_ID'] = "kai-ai-f63c8"
```
Comment (hash2004) [inline app/api/tests/test_tools.py:22]: Remove comments like these

File: app/api/tests/test_tools.py [hunk 4]
```diff
@@ -10,22 +10,25 @@
...
-    if old_creds:
-        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = old_creds
-    if old_env_type:
-        os.environ['ENV_TYPE'] = old_env_type
-    if old_project_id:
-        os.environ['PROJECT_ID'] = old_project_id
+    os.environ['GOOGLE_API_KEY'] = os.getenv('GOOGLE_API_KEY')
+
+    print('@@@@@ ENV')
+    print(os.environ['GOOGLE_API_KEY'])
```
Comment (hash2004) [inline app/api/tests/test_tools.py:20]: Printing env variables is not best practice.

**Review by hash2004 — CHANGES_REQUESTED** [review]

**Comment by AaronSosaRamos** [comment]
Closed as this issue was solved in the `Develop` branch

**Pull request closed without merging**